tests/test_sparse_kpca.py

```diff
@@ -0,0 +1,142 @@
+import unittest
+import warnings
+
+import numpy as np
+from sklearn import exceptions
+from sklearn.datasets import load_boston
+from sklearn.utils.validation import check_is_fitted
+
+from skcosmo.sparsified import SparseKPCA
+
+
+def _linear_kernel(X, Y=None):
+    if Y is None:
+        Y = X.copy()
+    return X @ Y.T
+
+
+class SparseKPCA_tests(unittest.TestCase):
+    def rel_error(self, A, B):
+        return np.linalg.norm(A - B) ** 2.0 / np.linalg.norm(A) ** 2.0
+
+    def setUp(self):
+        X, y = load_boston(return_X_y=True)
+        self.X = X
+        self.Y = y
+
+        self.error_tol = 1e-3
+        self.rounding = -int(round(np.log10(self.error_tol)))
+
+        self.kernels = ["linear", "poly", "rbf", "cosine", _linear_kernel]
+        self.n_active = 10
+
+    def check_kernel_types(self, kernel):
+        skpca = SparseKPCA(
+            n_components=2, tol=1e-12, n_active=self.n_active, kernel=kernel
+        )
+        _ = skpca.fit_transform(self.X)
```

Comment:
maybe even do the test by adding zero features to `self.X` such that you can test if the zero features are reduced out by KPCA
```py
X_expanded = np.hstack( (self.X, np.zeros((self.X.shape[0], 10))) )
skpca = SparseKPCA(
    n_components=self.X.shape[1], tol=1e-12, n_active=self.X.shape[0], kernel='linear', center=False
)
X_skpca = skpca.fit_transform(X_expanded)
K_skpca = X_skpca @ X_skpca.T
K = self.X @ self.X.T
# Since X_skpca = X @ U, then K_skpca = X_skpca @ X_skpca.T = X @ U @ U.T @ X.T = X @ X.T
self.assertTrue(np.allclose(K_skpca, K))